builtin/k8s/platform.go

```diff
@@ -125,6 +128,165 @@ func (p *Platform) getClientset() (*clientsetInfo, error) {
 	}, nil
 }
 
+func (p *Platform) resourceDeploymentStatus(
+	ctx context.Context,
+	log hclog.Logger,
+	sg terminal.StepGroup,
+	deploymentState *Resource_Deployment,
+	clientset *clientsetInfo,
+	sr *resource.StatusResponse,
+) error {
+	s := sg.Add("Checking status of the kubernetes deployment resource...")
+	defer s.Abort()
+
+	namespace := p.config.Namespace
+	if namespace == "" {
+		namespace = clientset.Namespace
+	}
+
+	// Get deployment status
+
+	deploymentResource := sdk.StatusReport_Resource{
+		Platform:            platformName,
+		Type:                "deployment",
+		CategoryDisplayHint: sdk.ResourceCategoryDisplayHint_INSTANCE_MANAGER,
+	}
+	sr.Resources = append(sr.Resources, &deploymentResource)
+
+	deployResp, err := clientset.Clientset.AppsV1().Deployments(namespace).Get(ctx, deploymentState.Name, metav1.GetOptions{})
+	if deployResp == nil {
+		return status.Errorf(codes.FailedPrecondition, "kubernetes deployment response cannot be nil")
+	} else if err != nil {
+		if !errors.IsNotFound(err) {
+			return status.Errorf(codes.FailedPrecondition, "error getting kubernetes deployment %s: %s", deploymentState.Name, err)
+		} else {
+			deploymentResource.Name = deploymentState.Name
+			deploymentResource.Health = sdk.StatusReport_MISSING
+			deploymentResource.HealthMessage = sdk.StatusReport_MISSING.String()
+
+			// Continue on with getting statuses for the rest of our resources
+		}
+	} else {
```

Comment:
What I don't love about structuring it that way is that it's hard to see that the only way to get to the part with `deploymentResource.Name = deploymentState.Name` is by having a not-found error. 

I updated the error message with your suggestion!

Reply:
That's fair, I suppose you could get the results of `errors.IsNotFound` and then use that in both of those checks, or just  document that condition but I see what you're saying. What I like about this refactoring is you have less to think about instead of tracking 2 or 3 levels of `if` conditions e.g. "if the response is not nil and the error is not nil and the error is not a not-found error, return ..."